src/extension.ts

```diff
@@ -26,8 +53,113 @@ export function activate(context: vscode.ExtensionContext) {
     )
   );
 
+  // Automatically open Live Interview + Sidebar
   vscode.commands.executeCommand("hatchways-live-interviewing.welcome");
   vscode.commands.executeCommand("hatchways-sidebar.focus");
+
+  // When user submitted their name
+  socket.on(USER_READY, (message) => {
+    vscode.window.showInformationMessage(message);
+  });
+
+  // When user left
+  socket.on(DISCONNECT, (message) => {
+    vscode.window.showInformationMessage(message);
+  });
+
+  // When user open a file
+  let disposableCurrFileDecorationProviders: any = [];
+  socket.on(FILE_CLICK, (value) => {
+    for (const d of disposableCurrFileDecorationProviders) {
+      d.dispose();
+    }
+
+    // For each online user, register a FileDecorationProvider
+    disposableCurrFileDecorationProviders = [];
+    for (const userId in value["allOnlineUsers"]) {
+      const currFileDecorationProvider = new FileDecorationProvider(
+        socket,
+        value,
+        userId
+      );
+      disposableCurrFileDecorationProviders.push(currFileDecorationProvider);
+    }
+  });
+
+  // When user click on a line
+  let disposableCursorDecorations: any = {};
+  socket.on(CURSOR_MOVE, (data) => {
+    const { userPerformingThisAction, cursorPosition, allOnlineUsers } = data;
+
+    // Remove previous cursor because the user is moving onto a new file now.
```

Comment:
This is moving on to a new line actually right? Do we need the same handling for when opening a new file?

Reply:
Actually when user open a new file, VSCode extension automatically move their cursor to 0,0 position so we don't need to worry about manually moving it